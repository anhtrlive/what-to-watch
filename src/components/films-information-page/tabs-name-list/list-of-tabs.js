import React from 'react';
import PropTypes from 'prop-types';
import {Link} from 'react-router-dom';

const ListOftabs = (props) => {
  const {activeTab} = props;

  return <nav className="movie-nav movie-card__nav">
    <ul className="movie-nav__list">
      {[`Overview`, `Details`, `Review`].map((el, index) => {
        const liClassName = (el.toLowerCase() === activeTab) ?
          `movie-nav__item movie-nav__item--active` :
          `movie-nav__item`;

        return <li className={liClassName} key={(el + index).toString()}
          style={{cursor: `pointer`}}>
          <Link to={el.toLowerCase()} className="movie-nav__link">{el}</Link>
        </li>;
      })}
    </ul>
  </nav>;
};

ListOftabs.propTypes = {
  activeTab: PropTypes.string.isRequired,
};

export {ListOftabs};
