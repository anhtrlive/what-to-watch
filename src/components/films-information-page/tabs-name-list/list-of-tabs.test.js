import React from 'react';
import Enzyme from 'enzyme';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import {BrowserRouter} from 'react-router-dom';
import {ListOftabs} from './list-of-tabs';

Enzyme.configure({adapter: new Adapter()});

const activeTab = `Details`;

it(`button plus list`, () => {
  const tree = renderer.create(
      <BrowserRouter>
        <ListOftabs
          activeTab={activeTab}
        />
      </BrowserRouter>).toJSON();

  expect(tree).toMatchSnapshot();
});
