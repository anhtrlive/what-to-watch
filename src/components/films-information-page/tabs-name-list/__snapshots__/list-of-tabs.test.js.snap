// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button plus list 1`] = `
<nav
  className="movie-nav movie-card__nav"
>
  <ul
    className="movie-nav__list"
  >
    <li
      className="movie-nav__item"
      style={
        Object {
          "cursor": "pointer",
        }
      }
    >
      <a
        className="movie-nav__link"
        href="/overview"
        onClick={[Function]}
      >
        Overview
      </a>
    </li>
    <li
      className="movie-nav__item"
      style={
        Object {
          "cursor": "pointer",
        }
      }
    >
      <a
        className="movie-nav__link"
        href="/details"
        onClick={[Function]}
      >
        Details
      </a>
    </li>
    <li
      className="movie-nav__item"
      style={
        Object {
          "cursor": "pointer",
        }
      }
    >
      <a
        className="movie-nav__link"
        href="/review"
        onClick={[Function]}
      >
        Review
      </a>
    </li>
  </ul>
</nav>
`;
